src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -205,6 +205,14 @@ private void SetVerb()
                         SetCurlOption(CURLoption.CURLOPT_COPYPOSTFIELDS, string.Empty);
                     }
                 }
+                else
+                {
+                    SetCurlOption(CURLoption.CURLOPT_CUSTOMREQUEST, _requestMessage.Method.Method);
+                    if (_requestMessage.Content != null)
+                    {
+                        SetCurlOption(CURLoption.CURLOPT_UPLOAD, 1L);
```

Comment:
> CUROPT_UPLOAD is not relevant for POST. And for HEAD we overrode it with CURLOPT_NOBODY

Sure, I meant for DELETE, TRACE, OPTIONS, GET, and any other custom methods.

If uploading supplied Content is the right thing to do in those cases, even if some servers might reject it, then this looks fine.  I just want to verify that's the desired behavior / matches what WinHttpHandler does.

From David's response, sounds like at least TRACE should be special cased not to send it, and then we can figure out exactly what to do with the others when David adds additional tests here.